docs/usage.adoc

```diff
@@ -114,9 +116,171 @@ Similar to disks, volumes can also be private or shared among nodes. Private vol
 
 A `size` and `class` params can be specified for every volume. If you don't specify them, the default values will be used. To specify default values use the <<installation#values,`values.yaml`>> file during the installation stage.
 
-==== sidecars & initContainers
-https://kubernetes.io/docs/concepts/workloads/pods/init-containers/[Init containers]
-  and sidecars can be added to the cluster.
+==== initContainers
+
+In the context of Xp7Deployment, `initContainers` are containers launched before the XP container itself, as part of the deployment lifecycle. This mechanism is particularly useful to set up the environment or parameters for the main application or to ensure all prerequisites are in place before the actual application starts.
+
+One important thing to remember is that, if there are several initContainers defined, they will be launched sequentially, following the order in the configuration. If any of the `initContainers` fail, Kubernetes will keep retrying until it succeeds.
+
+[source,yaml]
+----
+  ...
+      initContainers:
+        - name: init-db
+          image: db-init-image:latest
+          command: ["sh", "-c"]
+          args: ["until nslookup mydatabase; do echo waiting for mydatabase; sleep 2; done"]
+          env:
+            - name: CHECK_INTERVAL
+              value: "2"
+          mounts:
+            - name: init-scripts
+              mountPath: /scripts
+  ...
+----
+
+An `initContainer` block in `Xp7Deployment` has several possible fields:
+
+- `name`:The name of the `initContainer`
+- `image`: The image to use for the `initContainer`
+- `command`: The command to run in the `initContainer`
+- `args`: The arguments for the command in `initContainer`
+- `env`: Specify environment variables here. See <<env>> for more information.
+- `ports`: A list of port objects with properties `name`, `containerPort`, and `protocol`. This can be used to expose ports
+- `mounts`: A list of mount dirs, disks or volumes with properties `name`, `mountPath`, `subPath`.
+
+
+[cols="1,4"]
+|===
+| Property | Description
+
+|name
+|name of an actual `dir`, `disk` or `volume` name, defined in `Xp7Deployment` resource.
+
+|mountPath
+|It is the path to mount the `dir`, `disk` or `volume` in the container.
+
+|subPath
+|It is the path inside the `dir`, `disk` or `volume` to mount in the container. If not specified, the root path will be used.
+|===
+
+Read https://kubernetes.io/docs/concepts/workloads/pods/init-containers/[more] about the init containers.
+
+
+==== sidecars
+A sidecar is a utility container that is designed to support and enhance the functionality of the main XP application container. All containers within a pod share the same network stack, so sidecar containers can provide additional functionality without modifying the main container.
+
+For instance, a sidecar can be conceived to perform logging, monitoring, synchronization, and background data processing tasks.
```

Comment:
This is also too abstract.
Examples would be collect logs from xp log files, prometheus metrics, copying XP snapshots and backup blob store